File: two-pointers/lint-443/lint-443.js
/**
 * @param nums: an array of integer
 * @param target: An integer
 * @return: an integer
 */
const twoSum2 = function (nums, target) {
  nums.sort((a, b) => a - b);

  let left = 0;
  let right = nums.length - 1;

  let count = 0;
  while (left < right) {
    const value = nums[left] + nums[right];

    if (value <= target) {
      left += 1;
    } else {
      count += right - left;
      right -= 1;
    }
  }

  return count;
};

File: two-pointers/lint-443/lint-443.py
# https://www.lintcode.com/problem/two-sum-greater-than-target/description?_from=ladder&&fromId=1
"""
Time: O(nlogn+n)->O(nlogn)
Space: O(1)
"""


class Solution:
    """
    @param nums: an array of integer
    @param target: An integer
    @return: an integer
    """

    def twoSum2(self, nums, target):
        # write your code here
        nums.sort()
        left, right = 0, len(nums) - 1
        count = 0
        while left < right:
            value = nums[left] + nums[right]
            if value <= target:
                left += 1
            else:
                count += right - left
                right -= 1
        return count
